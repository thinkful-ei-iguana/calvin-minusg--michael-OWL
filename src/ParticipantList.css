participants-list {
  padding: 10px;
  position: relative;
  border: solid red 5px;
  margin-right: 82%;
}

.session-participant .participant-status.participant-status__green:after {
  background-color: #64d481;
}

.session-participant .participant-status.participant-status__gray:after {
  background-color: #d0d2d5;
}

.session-participant .user-avatar {
  margin-top: 5px;
}

.user-avatar {
  display: inline-block;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  font-size: 10px;
}

.side-bar {
  display: inline-block;
  position: absolute;
  width: 330px;
  height: calc(100vh - 75px);
  top: 0;
  left: -330px;
  font-size: 16px;
  border-right: 1px solid #eeeeef;
  text-align: left;
  transition: transform 0.19s ease-in-out;
}

.session {
  height: 100vh;
  text-align: right;
}

body,
html {
  height: 100%;
  background-color: #fff;
  color: #404853;
  font-family: Proxima Nova W01, Avenir Next W01, Avenir Next, sans-serif;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  line-height: 1.875;
  -webkit-font-smoothing: antialiased;
}

body {
  line-height: 1;
}

html {
  color: -internal-root-color;
}

::selection {
  background: #6799ff;
  color: #f9fbff;
}
