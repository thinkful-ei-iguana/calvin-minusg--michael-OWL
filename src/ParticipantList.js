import React from "react";
import "./ParticipantList.css";
import Participant from "./Participant";

export default function participantList(props) {
  let newProps = props.participants.filter(
    participant => participant.inSession === true
  );
  return (
    <div className="participants-list">
      {newProps.map(participant => (
        <Participant
          key={participant.id}
          name={participant.name}
          avatar={participant.avatar}
          inSession={participant.inSession}
          onStage={participant.onStage}
        />
      ))}
    </div>
  );
}
