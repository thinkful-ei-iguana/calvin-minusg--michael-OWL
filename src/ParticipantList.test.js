import React from 'react';
import ParticipantList from './ParticipantList';
import ReactDOM from 'react-dom';
import STORE from './STORE';
import renderer from 'react-test-renderer';

it('renders without crashing', () =>{
  const div = document.createElement('div');
  ReactDOM.render(<ParticipantList participants = {STORE} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<ParticipantList participants = {STORE}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();  
  });