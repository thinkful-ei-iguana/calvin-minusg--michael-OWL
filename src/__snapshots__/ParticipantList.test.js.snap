// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="participants-list"
>
  <div
    className="session-participant"
  >
    <img
      alt="Koren Templeton"
      className="user-avatar"
      src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Koren Templeton
      </h6>
      <div
        className="onStage-true"
      />
      <div
        className="onStage-true"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Caty Flucker"
      className="user-avatar"
      src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Caty Flucker
      </h6>
      <div
        className="onStage-false"
      />
      <div
        className="onStage-true"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Frank Runciman"
      className="user-avatar"
      src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Frank Runciman
      </h6>
      <div
        className="onStage-false"
      />
      <div
        className="onStage-true"
      />
    </div>
  </div>
  <div
    className="session-participant"
  >
    <img
      alt="Ashla Attwool"
      className="user-avatar"
      src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
    />
    <div
      className="participant-info"
    >
      <h6
        className="participant-name"
      >
        Ashla Attwool
      </h6>
      <div
        className="onStage-true"
      />
      <div
        className="onStage-true"
      />
    </div>
  </div>
</div>
`;
